lua/lspconfig/hls.lua

```diff
@@ -5,7 +5,11 @@ configs.hls = {
   default_config = {
     cmd = { 'haskell-language-server-wrapper', '--lsp' },
     filetypes = { 'haskell', 'lhaskell' },
-    root_dir = util.root_pattern('*.cabal', 'stack.yaml', 'cabal.project', 'package.yaml', 'hie.yaml'),
+    root_dir = function (fname)
+      local root_dir = util.root_pattern('*.cabal', 'stack.yaml', 'cabal.project', 'package.yaml', 'hie.yaml')(fname)
+      return root_dir
+        or vim.fn.getcwd()
```

Comment:
You probably want to use `util.path.dirname(fname)` instead of `vim.fn.getcwd()`.